### ISSUE TITLE
Rewards from Image Classification ?

### ISSUE THREAD

**Issue by shrirammoni** [Unity-Technologies/ml-agents#1105]
Hi,

Is there any way to get rewards from image classification results?

**Comment by vincentpierre** [comment]
Hi, 
As of today, ML-Agents use case is around training behaviors. Our algorithms use reinforcement learning or behavior cloning. 
If what you want is use an image classification model inside Unity, you could try to use TensorFlowSharp directly on a pre-trained model.
If what you want is to collect data relevant to train an image classifier, you could use a visual observation containing the image and a vector observation as the label of the image, this would allow you to collect the data you need.

To answer you question directly, if you have the visual observation be your image to classify and the vector action be the predicted label, you could give a positive reward when the predicted label matches the target label. You would set the classifying agent to done every few steps. Bear in mind that this is not what PPO was meant to do and you would need to do hyperparameter tuning (gamma must be 0 for instance).
I hope this helps.

**Comment by shrirammoni** [comment]
Hey @vincentpierre  thanks for the direction. In my case  there will multiple agents will be in the environment, which will have the ability to move around the environment. There will be some random points in the environment where the agents will go and sit(some kind of pattern formation) and then the screenshot is sent to image classifier where the image will be classified against an trained image classifier model and the result is given as reward to the agents. How to approach that?

**Comment by vincentpierre** [comment]
If I understand correctly, you want your agents to move around in the scene and place themselves to form an image. I have never thought of such a task and I think this is an interesting challenge. I would first try with very simple images and task and then gradually increase the difficulty. I think the inputs should be the target image, the current image, and information specific to the agent (position, color, etc...). I would then probably reward the agents at each time step for how close the target and screenshot are. I hope this helps.

**Comment by xiaomaogy** [comment]
@vincentpierre

**Comment by vincentpierre** [comment]
Thank you for the discussion. We are closing this issue due to inactivity. Feel free to reopen it if you’d like to continue to discussion though.
